app/controllers/RepresentativeDanImportVATController.scala

```diff
@@ -59,38 +60,46 @@ class RepresentativeDanImportVATController @Inject()(identify: IdentifierAction,
         backLink
       ))),
       dan => {
-        val previousVATAccountNumber = request.userAnswers.get(AdditionalDefermentNumberPage).getOrElse(dan.accountNumber)
-        val previousVATAccountType = request.userAnswers.get(AdditionalDefermentTypePage).getOrElse(dan.danType)
-        if (dan.accountNumber != previousVATAccountNumber || dan.danType != previousVATAccountType) {
+        val dutyAccountNumber = request.userAnswers.get(DefermentAccountPage)
+        if (dutyAccountNumber.contains(dan.accountNumber)) {
+          val form = formProvider().fill(RepresentativeDan(dan.accountNumber, dan.danType))
+            .withError(FormError("accountNumber", "repDan.error.input.sameAccountNumber"))
+          Future.successful(Ok(view(form, backLink)))
+        } else {
 
-          val authorityFiles = request.userAnswers.get(UploadAuthorityPage).getOrElse(Seq.empty).filterNot(_.dutyType == Vat)
+          val previousVATAccountNumber = request.userAnswers.get(AdditionalDefermentNumberPage).getOrElse(dan.accountNumber)
+          val previousVATAccountType = request.userAnswers.get(AdditionalDefermentTypePage).getOrElse(dan.danType)
+          if (dan.accountNumber != previousVATAccountNumber || dan.danType != previousVATAccountType) {
```

Comment:
these two are only used in the if condition below, could call a helper method passing user answers that would return a boolean and just call that in the if statement